Shared/Objects/DeviceProfileBuilder.swift

```diff
@@ -44,105 +17,166 @@ class DeviceProfileBuilder {
     }
 
     public func buildProfile() -> DeviceProfile {
```

Comment:
I've started to look at this in depth and I'll try to answer as many questions as I can but at least code-wise for now, let's turn this into:

`static func buildProfile(for type: VideoPlayerType, maxBitrate: Int)`

where then the type is switched on and each type has a private function that builds their respective profile. We can also just turn this into an `enum` to keep it namespaced.

I can't remember if leaving out the max bitrates would have the same effect as having no max (which is why we have the high temp max bitrate) so if that is true, just change to `maxBitrate: Int? = nil`.

Reply:
AFAIK: Leaving out the bitrate values in the `DeviceProfile` parameter will result in the Jellyfin server selecting it's default bitrate values, which are defined here: https://github.com/jellyfin/jellyfin/blob/2920611ffc206d845563637c4a865bf3f02d1374/MediaBrowser.Model/Dlna/DeviceProfile.cs#L132-L150

Reply:
@LePips I have change the function to static and added the parameters and a switch. The switch and respective private functions are not whole; they help build the profile, but they themselves don't return a full profile since many of the conditions are shared. If you would prefer the shared ones to be repeated in each individual function so the functions are more self-sufficient, let me know. (I am not sure if the current way is any cleaner than before code-wise, and it is very possible I misinterpretted what you are looking for!)